### PR TITLE
fix(all-in-one): Run compactor in all-in-one mode

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
While working on https://github.com/influxdata/EAR/issues/4363 I wanted to test a potential workaround that involved rewriting data and letting the compactor rewrite files. To do so I wanted to use the all-in-one mode. It turns out that all-in-one mode starts a compactor service, but it is configured in such a way that it can't do anything (it has a 0 second job timeout and thus all compactions timeout immediately 🤦 

Here is what happens if I do `cargo run` today:
```
2023-07-11T19:25:16.307170Z  INFO compactor_scheduler::local_scheduler::partitions_source::catalog_to_compact: Fetching partitions to consider for compaction minimum_time="2023-07-11T18:55:16.307154+00:00" maximum_time="" last_maximum_time="1970-01-01T00:00:00+00:00"
2023-07-11T19:25:16.308621Z  INFO compactor::components::partitions_source::logging: Fetch partitions n_partitions=6
2023-07-11T19:25:17.308224Z  INFO compactor::driver: compact partition partition_id=642
2023-07-11T19:25:17.308706Z ERROR compactor::components::partition_done_sink::logging: Error while compacting partition e=timeout without making any progress kind="timeout" partition_id=642
2023-07-11T19:25:17.310835Z  INFO compactor::driver: compacted partition partition_id=642
2023-07-11T19:25:18.309865Z  INFO compactor::driver: compact partition partition_id=641
2023-07-11T19:25:18.310525Z ERROR compactor::components::partition_done_sink::logging: Error while compacting partition e=timeout without making any progress kind="timeout" partition_id=641
2023-07-11T19:25:18.313789Z  INFO compactor::driver: compacted partition partition_id=641
2023-07-11T19:25:19.311659Z  INFO compactor::driver: compact partition partition_id=640
2023-07-11T19:25:19.312567Z ERROR compactor::components::partition_done_sink::logging: Error while compacting partition e=timeout without making any progress kind="timeout" partition_id=640
2023-07-11T19:25:19.315679Z  INFO compactor::driver: compacted partition partition_id=640
2023-07-11T19:25:20.313134Z  INFO compactor::driver: compact partition partition_id=639
2023-07-11T19:25:20.313791Z ERROR compactor::components::partition_done_sink::logging: Error while compacting partition e=timeout without making any progress kind="timeout" partition_id=639
2023-07-11T19:25:20.315929Z  INFO compactor::driver: compacted partition partition_id=639
2023-07-11T19:25:21.314713Z  INFO compactor::driver: compact partition partition_id=643
2023-07-11T19:25:21.315736Z ERROR compactor::components::partition_done_sink::logging: Error while compacting partition e=timeout without making any progress kind="timeout" partition_id=643
2023-07-11T19:25:21.318984Z  INFO compactor::driver: compacted partition partition_id=643
```

# Changes
1. Give compaction jobs a 30 minute timelimit in all in one mode
2. Allow more than one compaction job to run (target 1/2 of the number of cores)

Reminder that the all-in-one mode is designed to not be super tunable the way the other services are

### CODE REVIEW

File: influxdb_iox/src/commands/run/all_in_one.rs [hunk 1]
```diff
@@ -436,6 +435,15 @@ impl Config {
             catalog_dsn.dsn = Some(dsn);
         };
 
+        // TODO: make num_threads a parameter (other modes have it
+        // configured by a command line)
+        let num_threads =
```
Comment (alamb) [inline influxdb_iox/src/commands/run/all_in_one.rs:440]: I am sure these settings are non optimal for many different situations, but I think it is good enough for now

File: influxdb_iox/src/commands/run/all_in_one.rs [hunk 2]
```diff
@@ -595,20 +608,23 @@ pub async fn command(config: Config) -> Result<()> {
...
+        StorageId::from("iox_scratchpad"),
+    );
     let exec = Arc::new(Executor::new_with_config(ExecutorConfig {
         num_threads,
         target_query_partitions: num_threads,
-        object_stores: HashMap::from([(
-            parquet_store_real.id(),
-            Arc::clone(parquet_store_real.object_store()),
-        )]),
+        object_stores: [&parquet_store_real, &parquet_store_scratchpad]
```
Comment (alamb) [inline influxdb_iox/src/commands/run/all_in_one.rs:624]: the scratchpad was also not registered with the executor, which meant even after I fixed the timeout problem the compactor still failed 🤦

File: influxdb_iox/src/commands/run/all_in_one.rs [hunk 3]
```diff
@@ -487,15 +495,15 @@ impl Config {
...
+            compaction_df_concurrency: compactor_concurrency,
+            compaction_partition_scratchpad_concurrency: compactor_concurrency,
+            query_exec_thread_count: Some(num_threads),
             exec_mem_pool_bytes,
-            max_desired_file_size_bytes: 30_000,
+            max_desired_file_size_bytes: 100 * 1024 * 1024, // 100 MB
             percentage_max_file_size: 30,
             split_percentage: 80,
-            partition_timeout_secs: 0,
+            partition_timeout_secs: 30 * 60, // 30 minutes
```
Comment (wiedld) [inline influxdb_iox/src/commands/run/all_in_one.rs:506]: 🤦🏼‍♀️

Comment (alamb) [reply influxdb_iox/src/commands/run/all_in_one.rs:506]: Yeah, that was totally my bad for not understanding what this parameter was for

**Review by Joe-Blount — APPROVED** [review]
I wonder if the concurrency was set to 1 for ease of debugging?  Maybe thinking this would be used for debugging broken situations that compactor can't handle, so reduce concurrency so its easier to look at and reason about?
regardless, its easy enough to change the concurrency if/when its helpful.

**Comment by alamb** [comment]
> I wonder if the concurrency was set to 1 for ease of debugging? Maybe thinking this would be used for debugging broken situations that compactor can't handle, so reduce concurrency so its easier to look at and reason about?
regardless, its easy enough to change the concurrency if/when its helpful.

I am pretty sure I set it to one when I initially wrote `all-in-one` because I didn't know any better

I agree that using all-in-one for compactor debugging might be useful. Maybe it is worth a quick sync sometime to talk over the options

**Review by wiedld — APPROVED** [review]

**Comment by alamb** [comment]
> I agree that using all-in-one for compactor debugging might be useful. Maybe it is worth a quick sync sometime to talk over the options

Update here is that we had a sync and basically if @wiedld  and @Joe-Blount find the all in one mode useful for debugging the compactor, they should feel free to adapt it to their needs

**Pull request merged**